pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java

```diff
@@ -1118,4 +1118,9 @@ public void updateBrokerSubscriptionDispatchRate() {
         topicPolicies.getSubscriptionDispatchRate().updateBrokerValue(
             subscriptionDispatchRateInBroker(brokerService.pulsar().getConfiguration()));
     }
+
+    public void updateBrokerPublishRate() {
+        topicPolicies.getPublishRate().updateBrokerValue(
+            publishRateInBroker(brokerService.pulsar().getConfiguration()));
```

Comment:
When broker level topic publish rate is updated, all topic's publish rate is updated. 
Only those topic which topic level publish rate is null should be updated in this case.

Reply:
We don't need to check other level settings with `HierarchyTopicPolicies`

Reply:
Yes. But we use the `topicPolicies` is a `HierarchyTopicPolicies` type, which will update value through topic-namespace-broker hierarchy. For details : `org.apache.pulsar.common.policies.data.PolicyHierarchyValue#updateBrokerValue` @wangjialing218